src/web/app/components/extension-confirm-modal/extension-confirm-modal.component.spec.ts

```diff
@@ -141,16 +142,168 @@ describe('ExtensionConfirmModalComponent', () => {
   it('use ngOnInit to initialise', () => {
     component.selectedStudents = [studentModel3];
     component.selectedInstructors = [instructorModel3];
+    const setStudentTableDataSpy = jest.spyOn(component, 'setStudentTableData');
+    const setInstructorTableDataSpy = jest.spyOn(component, 'setInstructorTableData');
     component.ngOnInit();
     component.extensionTimestamp = testFeedbackSession.submissionEndTimestamp;
     fixture.detectChanges();
     expect(fixture).toMatchSnapshot();
+    expect(setStudentTableDataSpy).toHaveBeenCalled();
+    expect(setInstructorTableDataSpy).toHaveBeenCalled();
+  });
+
+  it('ngOnInit to initialise with empty arrays', () => {
+    component.selectedStudents = [];
+    component.selectedInstructors = [];
+    const setStudentTableDataSpy = jest.spyOn(component, 'setStudentTableData');
+    const setInstructorTableDataSpy = jest.spyOn(component, 'setInstructorTableData');
+    component.ngOnInit();
+    component.extensionTimestamp = testFeedbackSession.submissionEndTimestamp;
+    fixture.detectChanges();
+    expect(fixture).toMatchSnapshot();
+    expect(setStudentTableDataSpy).not.toHaveBeenCalled();
+    expect(setInstructorTableDataSpy).not.toHaveBeenCalled();
+  });
+
+  it('test setStudentRowData', () => {
+    component.selectedStudents = [studentModel1, studentModel2];
+    component.setStudentRowData();
+    const expectedData = component.selectedStudents.map((studentData: StudentExtensionTableColumnModel) => {
+    const rowData: SortableTableCellData[] = [
+        {
+          value: studentData.sectionName,
+        },
+        {
+          value: studentData.teamName,
+        },
+        {
+          value: studentData.name,
+        },
+        {
+          value: studentData.email,
+        },
+        {
+          value: studentData.extensionDeadline,
+          displayValue: component.dateDetailPipe.transform(
+          studentData.extensionDeadline,
+          component.feedbackSessionTimeZone),
+        },
+      ];
```

Comment:
this is generally not a good practice as we are essentially copying the code from the function to test the function, we should instead be manually typing out what we expect, for e.g.

const expectedData = [[...student1Data], [...student2Data]]

exactly as you did in setStudentColumnData